# üìÅ File: widgets/dialog_view_teacher.py

from PyQt6.QtWidgets import QDialog, QListWidgetItem
from PyQt6 import uic
from models.teacher import TeacherManager

class Dialog_View_Teacher(QDialog):
    def __init__(self, teacher_name):
        super().__init__()
        uic.loadUi("Ui/Dialog_View_Teacher.ui", self)

        self.setWindowTitle("Th√¥ng tin gi√°o vi√™n")
        
        self.teacher_mgr = TeacherManager()

        self.teacher = self.teacher_mgr.view_teacher(teacher_name)
        
        # ƒê·ªï d·ªØ li·ªáu v√†o c√°c label readonly
        self.name_label.setText(self.teacher.name)
        self.gender_label.setText(self.teacher.gioitinh)  # S·ª≠a l·∫°i
        self.age_label.setText(str(self.teacher.age))
        self.subject_label.setText(self.teacher.mon)       # S·ª≠a l·∫°i

        # N·∫øu gi√°o vi√™n n√†y l√† GVCN
        if self.teacher.gvcn:
            self.gvcn_label.setText(self.teacher.gvcn)
        else:
            self.gvcn_label.setText("Kh√¥ng ch·ªß nhi·ªám")

        # Hi·ªÉn th·ªã t·∫•t c·∫£ c√°c l·ªõp trong lop_day c·ªßa gi√°o vi√™n
        self.class_list.clear()
        for lop in self.teacher.lop_day:
            item = QListWidgetItem(f"{lop} - {self.teacher.mon}")
            self.class_list.addItem(item)
            
        # Disable t·∫•t c·∫£ ƒë·ªÉ kh√¥ng ch·ªânh s·ª≠a ƒë∆∞·ª£c
        self.name_label.setEnabled(False)
        self.gender_label.setEnabled(False)
        self.age_label.setEnabled(False)
        self.subject_label.setEnabled(False)
        self.gvcn_label.setEnabled(False)
        self.class_list.setEnabled(False)
